File: backend/internal/models/store.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Store struct {
	ID        uuid.UUID `gorm:"type:uuid;primaryKey"`
	Number    string    `gorm:"column:number"`
	Name      string    `gorm:"column:name"`
	LegalName string    `gorm:"column:legal_name"`

	AddressID uuid.UUID `gorm:"type:uuid;not null;column:address_id"`
	Address   Address   `gorm:"foreignKey:AddressID"`

	EstablishmentID uuid.UUID `gorm:"type:uuid;not null;column:establishment_id"`

	CreatedAt time.Time `gorm:"column:created_at"`
	UpdatedAt time.Time `gorm:"column:updated_at"`
}

File: frontend/src/types/store.ts
import type { EstablishmentResponse } from "./establishment";

export type StoreResponse = {
  ID: string;
  Number: string;
  Name: string;
  LegalName: string;
  AddressID: string;
  Address: {
    ID: string;
    Street: string;
    Number: string;
    City: string;
    State: string;
    ZipCode: string;
    CreatedAt: Date;
    UpdatedAt: Date;
  },
  EstablishmentID: string;
  Establishment: EstablishmentResponse;
  CreatedAt: Date;
  UpdatedAt: Date;
}

export type StorePreview = {
  id: string;
  name: string;
  legal_name: string;
  number: string;
  establishment_id: string; // Establishment Name
  address: {
    street: string;
    number: string;
    city: string;
    state: string;
    zip_code: string;
  }
}
